### PR TITLE
Return a Koalas series instead of pandas' in stats APIs at Koalas DataFrame

### LINKED ISSUE #1332: Return type consistency in docs
It is currently not transparent what the return type of a kolas dataframe / series is.

Take for example
[DataFrame.max()](https://koalas.readthedocs.io/en/latest/reference/api/databricks.koalas.DataFrame.max.html)

> Returns
maxscalar for a Series, and a Series for a DataFrame.

Returned type i receive is a pandas.Series

[Series.value_counts()](https://koalas.readthedocs.io/en/latest/reference/api/databricks.koalas.Series.value_counts.html)

> Returns
counts Series

Returned type i receive is a databricks.koalas.series.Series

I would like to suggest that it is made clear in the docs across all functions what the return type is in all cases

In both these cases it is unclear what type of series is returned

### PR DESCRIPTION
**Pull request by HyukjinKwon**
Currently, Koalas' statics functions such as count, sum, min, max, etc. returns a pandas Series.

Resolves #1332

### CODE REVIEW

File: databricks/koalas/tests/test_stats.py [hunk 1]
```diff
@@ -155,37 +156,40 @@ def test_stats_on_boolean_dataframe(self):
...
 
-        self.assertAlmostEqual(kser.var(), pser.var())
-        self.assertAlmostEqual(kser.std(), pser.std())
+        self.assert_eq(kser.var(), pser.var(), almost=True)
+        self.assert_eq(kser.std(), pser.std(), almost=True)
 
+    @unittest.skip(
+        "PySpark does not allow object type. "
+        "Should we support these case by casting to string? or just disallow?"
+    )
```
Comment (HyukjinKwon) [inline databricks/koalas/tests/test_stats.py]: I will make a followup after this PR.

File: databricks/koalas/frame.py [hunk 2]
```diff
@@ -491,19 +493,20 @@ def _reduce_for_stat_function(self, sfun, name, axis=None, numeric_only=False):
...
+                    new_column_labels.append(label)
 
             sdf = self._sdf.select(*exprs)
-            pdf = sdf.toPandas()
-
-            if self._internal.column_labels_level > 1:
-                pdf.columns = pd.MultiIndex.from_tuples(self._internal.column_labels)
 
-            assert len(pdf) == 1, (sdf, pdf)
+            # The data is expected to be small so it's fine to transpose/use default index.
```
Comment (ueshin) [inline databricks/koalas/frame.py]: How about adding a column like `F.lit(0).alias(SPARK_DEFAULT_INDEX_NAME)` and `index_map=[(SPARK_DEFAULT_INDEX_NAME, None)]` without default index?

File: databricks/koalas/frame.py [hunk 3]
```diff
@@ -491,19 +493,20 @@ def _reduce_for_stat_function(self, sfun, name, axis=None, numeric_only=False):
...
+                index_map=kdf._internal.index_map,
+                column_labels=new_column_labels,
+                column_label_names=self._internal.column_label_names,
+            )
 
-            row = pdf.iloc[0]
-            row.name = None
-            # TODO: return Koalas series.
-            return row  # Return first row as a Series
+            return _col(DataFrame(internal).transpose())
```
Comment (ueshin) [inline databricks/koalas/frame.py]: Should we use `with ks.option_context("compute.max_rows", None)` to avoid Spark job?

**Comment by HyukjinKwon** [comment]
I discussed offline with @ueshin. I am going to merge this.

**Pull request merged**